src/utils.js

```diff
@@ -379,7 +379,7 @@ module.exports = (function() {
         // format:json does not actually work, so we need to grab the photo ID from the response XML:
         // <?xml version="1.0" encoding="utf-8" ?>\n<rsp stat="ok">\n<photoid>.........</photoid>\n</rsp>\n
         var data = undefined;
-        if(body.indexOf('rsp stat="ok"')>-1) {
+        if(body && body.indexOf('rsp stat="ok"')>-1) {
```

Comment:
since we're doing a callback with possible error, it'll be better to use:
```
if(!body) {
  error = "No body found in response";
} else if (body.indexOf('rsp stat="ok"')>-1) {
  data = ...
}
callback(error, data);
```
That way the calling code can still switch paths on errors.